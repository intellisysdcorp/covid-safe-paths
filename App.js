import 'react-native-gesture-handler';

import * as Sentry from '@sentry/react-native';
import React, { useEffect, useReducer } from 'react';
import { MenuProvider } from 'react-native-popup-menu';
import SplashScreen from 'react-native-splash-screen';

import context from './app/components/DR/Reduces/context';
import { initialState, reducer } from './app/components/DR/Reduces/index';
import StatusBar from './app/components/DR/StatusBar';
import Colors from './app/constants/colors';
import { SENTRY_DNS } from './app/constants/DR/baseUrls';
import { Theme } from './app/constants/themes';
import Entry from './app/Entry';
import { FlagsProvider } from './app/helpers/Flags';
import VersionCheckService from './app/services/VersionCheckService';

Sentry.init({
  dsn: SENTRY_DNS,
  enableAutoSessionTracking: true,
  sessionTrackingIntervalMillis: 10000,
});

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
    VersionCheckService.start();
  }, []);

  const [state, setState] = useReducer(reducer, initialState);

  return (
    <context.Provider value={[state, setState]}>
      <StatusBar
        barStyle={'light-content'}
        backgroundColor={Colors.BLUE_RIBBON}
      />
      <FlagsProvider>
        <MenuProvider>
          <Theme use='default'>
            <Entry />
          </Theme>
        </MenuProvider>
      </FlagsProvider>
    </context.Provider>
  );
};

export default App;
