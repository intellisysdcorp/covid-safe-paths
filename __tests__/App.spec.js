/**
 * @format
 */

import 'react-native';
import 'isomorphic-fetch';

import { render } from '@testing-library/react-native';
import React from 'react';

import App from '../App';

jest.mock('../app/Entry', () => 'Entry');

jest.mock('@sentry/react-native', () => ({
  init: jest.fn(),
}));

it('renders correctly', () => {
  const { asJSON } = render(<App />);

  expect(asJSON()).toMatchSnapshot();
});
