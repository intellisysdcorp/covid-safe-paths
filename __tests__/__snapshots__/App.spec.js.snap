// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 44,
        },
        Object {
          "backgroundColor": "#0161F2",
        },
      ]
    }
  />
  <MenuProvider>
    <Entry />
  </MenuProvider>
</View>
`;
